Backpack/Nudger/Classes/BPKNudger.m

```diff
@@ -0,0 +1,297 @@
+        [self.topAnchor constraintEqualToAnchor:self.stackView.topAnchor],
+        [self.leadingAnchor constraintEqualToAnchor:self.stackView.leadingAnchor],
+        [self.stackView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor],
+        [self.stackView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
+        [self.label.widthAnchor constraintGreaterThanOrEqualToConstant:minWidth],
+    ]];
+    self.accessibilityTraits = UIAccessibilityTraitAdjustable;
+    self.isAccessibilityElement = YES;
+
+    [self updateDisplay];
+}
+
+#pragma mark - Properties
+
+- (UIStackView *)stackView {
+    BPKAssertMainThread();
+    if (_stackView) {
+        return _stackView;
+    }
+
+    UIStackView *stackView = [[UIStackView alloc] init];
+    stackView.axis = UILayoutConstraintAxisHorizontal;
+    stackView.alignment = UIStackViewAlignmentCenter;
+    stackView.spacing = BPKSpacingMd;
+
+    [stackView addArrangedSubview:self.decrementButton];
+    [stackView addArrangedSubview:self.label];
+    [stackView addArrangedSubview:self.incrementButton];
+
+    _stackView = stackView;
+
+    return _stackView;
+}
+
+- (BPKButton *)decrementButton {
+    BPKAssertMainThread();
+    if (_decrementButton) {
+        return _decrementButton;
+    }
+
+    BPKButton *decrementButton = [[BPKButton alloc] initWithSize:BPKButtonSizeDefault style:BPKButtonStyleSecondary];
+    [decrementButton setImage:[BPKIcon smallTemplateIconNamed:BPKSmallIconNameMinus]];
+
+    _decrementButton = decrementButton;
+    _decrementButton.accessibilityLabel = self.configuration.decrementButtonLabel;
+    [_decrementButton addTarget:self action:@selector(didTapDecrement:) forControlEvents:UIControlEventTouchUpInside];
+
+    return _decrementButton;
+}
+
+- (BPKNudgerLabel *)label {
+    BPKAssertMainThread();
+    if (_label) {
+        return _label;
+    }
+
+    BPKNudgerLabel *label = [[BPKNudgerLabel alloc] initWithFrame:CGRectZero];
+    label.text = @"0";
+
+    label.frameProvider = self;
+    _label = label;
+
+    return _label;
+}
+
+- (BPKButton *)incrementButton {
+    BPKAssertMainThread();
+    if (_incrementButton) {
+        return _incrementButton;
+    }
+
+    BPKButton *incrementButton = [[BPKButton alloc] initWithSize:BPKButtonSizeDefault style:BPKButtonStyleSecondary];
+    [incrementButton setImage:[BPKIcon smallTemplateIconNamed:BPKSmallIconNamePlus]];
+
+    _incrementButton = incrementButton;
+    _incrementButton.accessibilityLabel = self.configuration.incrementButtonLabel;
+    [_incrementButton addTarget:self action:@selector(didTapIncrement:) forControlEvents:UIControlEventTouchUpInside];
+
+    return _incrementButton;
+}
+
+#pragma mark - Actions
+
+- (void)didTapDecrement:(BPKButton *)sender {
+    [self decrementValue];
+}
+
+- (void)didTapIncrement:(BPKButton *)sender {
+    [self incrementValue];
+}
+
+
+#pragma mark - Setters / Logic
+
+- (void)decrementValue {
+    self.value -= self.stepValue;
+}
```

Comment:
Can a consumer hook up to these events with a delegate or something?

Reply:
Plan is to emit an event for `UIControlEventValueChanged`.

Reply:
That seems like it would be hard to debug?

Reply:
How do you mean? It's standard for `UIControl` subclasses. `UISlider` and `UIStepper` work the same way